driver/ppm_events.c

```diff
@@ -885,6 +885,28 @@ static struct socket *ppm_sockfd_lookup_light(int fd, int *err, int *fput_needed
 }
 */
 
+static void unix_socket_path(char *dest, const char *path)
+{
+	if (path[0] == '\0') {
+		/*
+		 * Extract from: https://man7.org/linux/man-pages/man7/unix.7.html
+		 * an abstract socket address is distinguished (from a
+		 * pathname socket) by the fact that sun_path[0] is a null byte
+		 * ('\0').  The socket's address in this namespace is given by
+		 * the additional bytes in sun_path that are covered by the
+		 * specified length of the address structure.
+		 */
+		snprintf(dest,
+			 UNIX_PATH_MAX,
+			 "@%s",
+			 path + 1);
+	} else {
+		dest = strncpy(dest,
+			       path,
+			       UNIX_PATH_MAX); /* we assume this will be smaller than (targetbufsize - (1 + 8 + 8)) */
```

Comment:
`strncpy` does not put a `\0` at the end of the string in case the specified length is reached. Do we need to handle this case here?

Reply:
As far as i can see, a NULL terminator is put in any case, after the function call (see lines below: 
`dest[UNIX_PATH_MAX - 1] = 0;`
Btw you helped me find a bug! I forgot to update `dest` pointer in one case.

Reply:
So this function, as written, may be a bit dangerous 🤔 as it also assumes that the destination buffer is at least `UNIX_PATH_MAX` bytes long. While the use is safe with the current caller, if someone would reuse this function somewhere else it'd be very easy to make a mistake. And in the kernel every mistake would be more dangerous ... Perhaps we could add the length as an argument and move the `dest[UNIX_PATH_MAX - 1] = 0;` line here.

Reply:
i like this! Thanks for the input!